File: java-vfs/src/main/java/emg/java/vfs/commands/Cat.java
package emg.java.vfs.commands;

import emg.java.vfs.filesystem.State;

public class Cat extends Command {
    private final String fileName;

    public Cat(String fileName) {
        this.fileName = fileName;
    }

    @Override
    public State apply(State state) {
        var wd = state.wd();
        var dirEntry = wd.findEntry(fileName);
        if (dirEntry == null || !dirEntry.isFile()) {
            return state.setMessage(fileName + ": no such File.");
        } else {
            return state.setMessage(dirEntry.asFile().contents());
        }
    }
}

File: scala-vfs/src/main/scala/emg/scala/vfs/commands/Cat.scala
package emg.scala.vfs.commands

import emg.scala.vfs.filesystem.State

class Cat(fileName: String) extends Command {
  override def apply(state: State): State = {
    val wd = state.wd
    val dirEntry = wd.findEntry(fileName)
    if (dirEntry == null || !dirEntry.isFile)
      state.setMessage(s"$fileName: no such File.")
    else
      state.setMessage(dirEntry.asFile.contents)
  }
}
